src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -1,62 +1,79 @@
-Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function(e) {
-    if (e.processedNewNodeConsoleNote) {
-        return
-    }
-    e.processedNewNodeConsoleNote = true
-    var label = e.getAttribute('label')
-    if (label != null) {
-        var html = e.innerHTML
-        var suffix = ' (' + label.escapeHTML() + ')';
-        if (html.indexOf(suffix)<0) {
-            e.innerHTML = e.innerHTML.replace(/.+/, '$&' + suffix) // insert before EOL
+/**
+ * The setTimeout() method calls a function or evaluates an expression after a specified number of milliseconds.
+ * It is used to prevent a browser freeze that might occur if a very big log is being evaluated.
+ */
+setTimeout(() => {
+    /**
+     * Specifies something to do when an element matching a CSS selector is encountered.
+     *
+     * @param {String} selector a CSS selector triggering your behavior
+     * @param {String} id combined with selector, uniquely identifies this behavior; prevents duplicate registrations
+     * @param {Number} priority relative position of this behavior in case multiple apply to a given element; lower numbers applied first (sorted by id then selector in case of tie); choose 0 if you do not care
+     * @param {Function} behavior callback function taking one parameter, a (DOM) {@link Element}, and returning void
+     */
+    Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function (e) {
+        var nodeId = e.getAttribute('nodeId')
+        var oid = e.getAttribute('nodeId')
+        var nodes = $$('.pipeline-new-node')
+        var enclosings = new Map() // id → enclosingId
+        var labels = new Map() // id → label
+
+        if (e.processedNewNodeConsoleNote) {
+            return
         }
-    }
-    var nodeId = e.getAttribute('nodeId')
-    var startId = e.getAttribute('startId')
-    if (startId == null || startId == nodeId) {
-        e.innerHTML = e.innerHTML.replace(/.+/, '$&<span class="pipeline-show-hide"> (<a href="#" onclick="showHidePipelineSection(this); return false">hide</a>)</span>')
-        // TODO automatically hide second and subsequent branches: namely, in case a node has the same parent as an earlier one
```

Comment:
I think this is already filed in JIRA somewhere.

Reply:
I removed this part, because it would probably make more sense to generate this inside the back-end.

Reply:
I should have clarified that my review comment was in reference to the removal of the one-line code comment, which is properly an RFE that should be tracked in Jira, not to behavioral changes in your patch.